usecases/replica/coordinator.go

```diff
@@ -273,7 +273,7 @@ func (c *coordinator[T]) Pull(ctx context.Context,
 
 					timer := time.NewTimer(nextBackOff)
 					select {
-					case <-ctx.Done():
+					case <-workerCtx.Done():
```

Comment:
If the parent `ctx` is cancelled, that will cancel all the associated `workerCtx`'s, right?

Reply:
yes, if the parent get's cancelled , it cancel it's grand/children 
> The [WithCancel](https://pkg.go.dev/context#WithCancel), [WithDeadline](https://pkg.go.dev/context#WithDeadline), and [WithTimeout](https://pkg.go.dev/context#WithTimeout) functions take a Context (the parent) and return a derived Context (the child) and a [CancelFunc](https://pkg.go.dev/context#CancelFunc). Calling the CancelFunc cancels the child and its children, removes the parent's reference to the child, and stops any associated timers. Failing to call the CancelFunc leaks the child and its children until the parent is canceled or the timer fires. The go vet tool checks that CancelFuncs are used on all control-flow paths.

w.r.t https://pkg.go.dev/context